### PR TITLE
feat: allow task name editing for v0.4.x

### PR DESCRIPTION
**Pull request by BobToninho**
Reference: https://github.com/alanvardy/tod/pull/245

Tests are missing, I just tested the feature with my account. I would like to have feedback on the code organization as I mainly focused on making things work (which was actually straightforward!). Should I add a method to `projects` for editing the tasks, instead of using `request` directly?

### CODE REVIEW

**Comment by BobToninho** [comment]
Also, bigger problem is on how to display the current content of the task. My idea was to print the current task name and make it editable from the user, but from a quick overview of the inquire lib it's not possible

**Review by alanvardy — COMMENTED** [review]
I'm pretty happy with this! Just a few comments around throwing exceptions.

Yeah, in general there is too much logic in the request module but I am going to leave that for a later refactor.

Thank you for the PR, once the CI is passing and comments addressed I will approve and merge unless @titoOdUA has anything extra :+1:

File: src/config.rs
Comment (BobToninho) [reply src/config.rs:229]: Sure! I missed that.

File: src/main.rs
Comment (BobToninho) [reply src/main.rs:243]: What about a message, like "The content is the same, no need to change it"? It's more informative to the user

Comment (alanvardy) [reply src/main.rs:243]: Even better!

**Comment by alanvardy** [comment]
> Also, bigger problem is on how to display the current content of the task. My idea was to print the current task name and make it editable from the user, but from a quick overview of the inquire lib it's not possible

I haven't tried it but `initial_value` might be what you are looking for
https://docs.rs/inquire/0.6.2/inquire/struct.Text.html

If that doesn't work you could use the Editor option as it allows predefined text
https://docs.rs/inquire/0.6.2/inquire/struct.Editor.html

File: src/main.rs
Comment (titoOdUA) [inline src/main.rs:232]: I see that there is a lot of work with the project entity. 

I might be wrong but isn't we now have some struct for projects?
@alanvardy I think it was one of the recent refactors, maybe too recent so it is not present here =)

Maybe we can fetch the project object, and from that point work with a strongly typed entity? Rather than use associated methods n times over and over.

I thought that we have the ability now to acquire Project objects. Then we can just use fields that are populated with related data (project.name, project.id, etc)

It is a minor thing and may be addressed in the future.
Other than that it is a good addition to the interface. =)

File: src/main.rs
Comment (titoOdUA) [inline src/main.rs:420]: I actually might be dumb today (had very little sleep), sorry if I'm blind and silly.

Where it comes into play? I am under the impression that in task_edit we receive tasks only from prompt.

**Review by titoOdUA — COMMENTED** [review]
I have some comments. In general, it looks good and useful! =)
One regarding task_edit, one regarding task arg.

File: src/main.rs
Comment (alanvardy) [reply src/main.rs:420]: Yeah, I double checked and I don't think that this argument is being used either, remove?

File: src/main.rs
Comment (alanvardy) [inline src/main.rs:243]: I think that everything from line 233 to 246 can be moved to the `projects`module and named something like "rename_item"

The reason being is that we want to keep the code in the main module as small as possible (it just gets the necessary values from matches) and then do the testing in the other modules where we won't have to construct the matches.

File: src/main.rs
Comment (alanvardy) [reply src/main.rs:232]: I haven't done that yet, but would like to in another refactor. Great idea!

File: src/main.rs
Comment (BobToninho) [reply src/main.rs:420]: Yeah, I added it in the beginning thinking the task could be passed as arg, but the implementation changed. I'll remove it!

**Comment by alanvardy** [comment]
@BobToninho @titoOdUA I'm going to hold off merging my work for a couple of days or until this gets merged because some of my refactors will conflict with this.

**Comment by BobToninho** [comment]
Great! Will finish this in the following days

**Review by alanvardy — COMMENTED** [review]
This is great, thank you!

I have PRs after this that will make it easy to test. Are you ready for me to merge?

**Comment by BobToninho** [comment]
> This is great, thank you!
> 
> I have PRs after this that will make it easy to test. Are you ready for me to merge?

So we'll add tests after those PRs? If yes, you can merge!

**Review by alanvardy — APPROVED** [review]
Awesome, thanks Roberto

**Pull request merged**
